base/namedtuple.jl

```diff
@@ -404,11 +407,35 @@ end
     types = diff_types(a, names)
     n = length(names)
     A = Vector{Any}(undef, n)
-    for i=1:n
+    for i = 1:n
         n = names[i]
         A[i] = getfield(a, n)
     end
-    _new_NamedTuple(NamedTuple{names, types}, (A...,))
+    _new_NamedTuple(NamedTuple{names,types}, (A...,))
+end
+
+"""
+    delete(a::NamedTuple, keys::Union{Symbol, Tuple{Vararg{Symbol}}})
+
+Construct a new named tuple from `a` by removing the named entries.
+
+```jldoctest
+julia> delete((a=1, b=2, c=3), :a)
+(b = 2, c = 3)
+julia> delete((a=1, b=2, c=3), :b)
+(a = 1, c = 3)
+julia> delete((a=1, b=2, c=3), (:c, :b))
+(a = 1,)
+```
+"""
+function delete(a::NamedTuple{an}, @nospecialize(key::Symbol)) where {an}
+    names = diff_names(an, (key,))
+    NamedTuple{names}(a)
+end
+
+function delete(a::NamedTuple{an}, @nospecialize(keys::Tuple{Vararg{Symbol}})) where {an}
```

Comment:
I know it's a pain, but there should probably be a serious conversation concerning when tuples of indices are used and how they're interpreted. We have tuples of indices for a single index on multidim arrays, but how should we refer to multiple indices on tuples and named tuples? There's a loss of information if we require it always be another subtype of `AbstractArray`